src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,80 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { CommentData } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  addComment: (newComment: CommentData) => void;
+  buttonAddLoading: boolean;
+  selectedPost: Post | null;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  addComment,
+  buttonAddLoading,
+  selectedPost,
+}) => {
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState(false);
+  const [bodyComment, setBodyComment] = useState('');
+  const [bodyCommentError, setBodyCommentError] = useState(false);
+
+  let newComment: CommentData;
+
+  const handleFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setNameError(false);
+    setEmailError(false);
+    setBodyCommentError(false);
+
+    let hasError = false;
+
+    if (name.trim().length === 0) {
+      setNameError(true);
+      hasError = true;
+    }
```

Comment:
You are repeating the same logic for validating the `name`, `email`, and `bodyComment`. You could create a helper function to avoid repetition. The helper function could take the value to be validated and the setter function for the corresponding error state as arguments.